addons/purchase_mrp/models/stock_move.py

```diff
@@ -22,7 +21,8 @@ def _get_price_unit(self):
         line = self.purchase_line_id
         kit_price_unit = line._get_gross_price_unit()
         cost_share = self.bom_line_id._get_cost_share()
-        return kit_price_unit * cost_share
+        price_unit_prec = self.env['decimal.precision'].precision_get('Product Price')
+        return float_round(kit_price_unit * cost_share / self.bom_line_id.product_qty, precision_digits=price_unit_prec)
```

Comment:
That's not correct because it does not include:
- `bom.product_qty` that can be `!= 1`
- sub-kits

(Do not hesitate to improve the test to include both)

Such situations already exist in the code, I think if you look for the use of `MrpBom.explode` you may find some examples